### ISSUE TITLE
Disabling updates

### ISSUE THREAD

**Issue by cemo** [zalando-incubator/kubernetes-on-aws#431]
Hi, 

I am usually following your resources for cross check my configuration. I also want to create immutable infrastructure but I have some doubts about configuration and coreos documentation. I wanted to open an issue to be sure about it.

CoreOS documentation states that `reboot_strategy: off` means `Do not reboot after updates are applied`. I understood from this it can apply updates in case a reboot. 

What I am suggesting is that: 

`reboot_strategy: etcd-lock`

 and 
```
    - name: update-engine.service
      mask: true
    - name: locksmithd.service
      mask: true
```
. I think that we need to disable `update engine` and `locksmithd` service and in case a reboot which can be caused from something else `etcd-lock` is best choice to a proper boot.

What do you think?

**Comment by mikkeloscar** [comment]
@cemo What do you want to achieve? To not update at all even if an update is ready?

Then I think you are right in your approach, but I think there currently is an issue with masking the services when using `cloud-config`. https://github.com/coreos/bugs/issues/1646#issuecomment-258480287

So switching to Container Linux Config and ignition is probably the right way to go to get this feature fully supported.

I'm also interested in looking into https://github.com/coreos/container-linux-update-operator to orchestrate updates. We didn't enable automatic updates yet, but we want to have it in the future.

It's great that you find our configs helpful!

**Comment by cemo** [comment]
I would like to disable updates completely even if update is ready. I am considering to bootstrap a new cluster in case a new kubernetes update to provide immutable servers.

Thanks for the pointing the bug. I was considering to move ignition completely. 

I am following your project for a long time. You are providing nice insights and clues for us since we don't have a team as you have. Thanks for providing this project open source.

**Comment by cemo** [comment]
It seems that the correct way is disabling only `update-engine.service` but this is causing some cryptic errors. Locksmith depends on update-engine somehow. This is causing some problems when I stop update-engine service. So I decided to go same way you did. I opened some issues in coreos bug tracker and in case an update I will let you know as well.

In this configuration if we have a restart after an update, CoreOS will be updated. However I could not find a better way. I will close this issue for now and might reopen in future to provide updates.
